pkg/storage/allocator_test.go

```diff
@@ -1913,6 +1924,79 @@ func TestAllocatorComputeAction(t *testing.T) {
 	}
 }
 
+func TestAllocatorComputeActionDisableStatsRebalance(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	ctx := context.Background()
+	stopper, g, _, a, _ := createTestAllocator( /* deterministic */ false)
+	defer stopper.Stop(ctx)
+	stores := []*roachpb.StoreDescriptor{
+		{
+			StoreID: 1,
+			Node:    roachpb.NodeDescriptor{NodeID: 1},
+			Capacity: roachpb.StoreCapacity{
+				RangeCount: 100,
+			},
+		},
+		{
+			StoreID: 2,
+			Node:    roachpb.NodeDescriptor{NodeID: 2},
+			Capacity: roachpb.StoreCapacity{
+				RangeCount: 90,
+			},
+		},
+		{
+			StoreID: 3,
+			Node:    roachpb.NodeDescriptor{NodeID: 3},
+			Capacity: roachpb.StoreCapacity{
+				RangeCount: 95,
+			},
+		},
+		{
+			StoreID: 4,
+			Node:    roachpb.NodeDescriptor{NodeID: 4},
+			Capacity: roachpb.StoreCapacity{
+				RangeCount: 15,
+			},
+		},
+	}
+	sg := gossiputil.NewStoreGossiper(g)
+	sg.GossipStores(stores, t)
+
+	zoneConfig := config.ZoneConfig{
+		NumReplicas:   3,
+		Constraints:   config.Constraints{Constraints: []config.Constraint{{Value: "us-east"}}},
+		RangeMinBytes: 0,
+		RangeMaxBytes: 64000,
+	}
+
+	for i := 0; i < 50; i++ {
+		desc := roachpb.RangeDescriptor{
+			RangeID: roachpb.RangeID(i),
+			Replicas: []roachpb.ReplicaDescriptor{
+				{
+					StoreID:   1,
+					NodeID:    1,
+					ReplicaID: 1,
+				},
+				{
+					StoreID:   2,
+					NodeID:    2,
+					ReplicaID: 2,
+				},
+				{
+					StoreID:   3,
+					NodeID:    3,
+					ReplicaID: 3,
+				},
+			},
+		}
+		action, _ := a.ComputeAction(ctx, zoneConfig, RangeInfo{Desc: &desc}, true)
```

Comment:
It'd be nice to run this for both possible input values of the `disableStatsBasedRebalance` parameter to verify that there's a difference in behavior (i.e. that the test is testing what we want it to).